### PR TITLE
feat: convert hub to nuxt project

### LINKED ISSUE #691: 🤔 Migrate to nuxt.js
Nuxt seems to be a fairly good high-convention option for vue projects... It'd be great to standardize our format to something highly usable to make it easy for other devs wanting to contribute to be able to jump in...

I found [this](https://dev.to/ngranja19/how-to-migrate-a-vuejs-project-to-nuxtjs-in-8-steps-41dm) guide a while back and started the migration locally, but I've not had a ton of luck getting it to actually run. 

I like that there's better configuration for specifying styles in one spot, as well as easier plugin separation/declaration and ordering instead of shifting around a bunch of code in one big file.

It might need to wait until we can migrate up to VueJS 4 so that stuff like the i18n can properly use the new Vue.use syntax changes (taking in an instance instead of the class)

TODO: 
- [x] timers load
- [x] #732
- [x] #690 
- [x] #733
- [x] #734
- [x] navbar links don't show the right color for the active page
- [x] #735
- [x] remap crowdin files, files moved rom src -> static
- [ ] reimplement initialWorldstateUpdater.js
- [x] set notifications back up, they're commented
- [x] reimplement updates notifications (probably a nuxt module?)

### PR DESCRIPTION
**Pull request by TobiTenno**
**Summary:**
Convert to nuxt-based app
should make it easier to maintain and keep up with vue changes in the long run and have a bit more of a stable community

`vue-mobile-detection` -> `@nuxt/device`
sentry modules -> `@nuxt/sentry` 
App.vue -> default layout

---
**Fixes issue (include link):**
closes #691 

- [x] #691 

---
**Mockups, screenshots, evidence:**

---

### CODE REVIEW

File: components/panels/ConstructionPanel.vue [hunk 1]
```diff
@@ -38,7 +43,8 @@
 </template>
 
 <script>
-import VueCircle from 'vue2-circle-progress';
+// import VueCircle from 'vue2-circle-progress';
```
Comment (TobiTenno) [inline components/panels/ConstructionPanel.vue]: need to delete that line
```suggestion
```

File: components/panels/ConstructionPanel.vue [hunk 2]
```diff
@@ -48,7 +54,8 @@ export default {
   components: {
     NoDataItem,
     HubPanelWrap,
-    VueCircle,
+    // VueCircle,
```
Comment (TobiTenno) [inline components/panels/ConstructionPanel.vue]: delete
```suggestion
```

File: static/sw.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+self.addEventListener('activate', function (e) {
+  self.registration.unregister()
+    .then(function () {
+      return self.clients.matchAll()
+    })
+    .then(function (clients) {
+      clients.forEach(client => client.navigate(client.url))
+    })
+})
```
Comment (TobiTenno) [inline static/sw.js]: need to delete this file

File: pages/vallis/map.vue [hunk 4]
```diff
@@ -307,5 +302,10 @@ export default {
       });
     });
   },
+  methods: {
+    track() {
+      this.$ga.page('/vallis/map');
+    },
+  },
```
Comment (TobiTenno) [inline pages/vallis/map.vue]: delete
```suggestion
```

File: layouts/default.vue [hunk 5]
```diff
@@ -4,38 +4,36 @@
...
+// import update from '@/mixins/update';
 export default {
-  name: 'app',
   components: {
     Navbar,
     Settings,
     About,
   },
-  mixins: [update],
+  // mixins: [update],
```
Comment (TobiTenno) [inline layouts/default.vue]: delete
```suggestion
```

File: layouts/default.vue [hunk 6]
```diff
@@ -4,38 +4,36 @@
...
+import { mapGetters } from 'vuex';
 import Navbar from '@/components/Navbar.vue';
 import Settings from '@/components/modalDialogs/Settings.vue';
 import About from '@/components/modalDialogs/About.vue';
-import themes from '@/assets/json/themes.json';
-import { mapGetters } from 'vuex';
-import update from '@/mixins/update';
-
+import themes from '@/static/json/themes.json';
+// import update from '@/mixins/update';
```
Comment (TobiTenno) [inline layouts/default.vue]: delete
```suggestion
```

**Comment by TobiTenno** [comment]
@Bitti09 I'd love your thoughts on these since  you have a bit more experience with nuxt than I do, and you've typically got some pretty good thoughts. I worry that i've got too much freeform vue paradigms in play with my config & stuff like static asset usage

**Review by EricSihaoLin — APPROVED** [review]

**Comment by EricSihaoLin** [comment]
I read through all the changes from a javascript standpoint, all of these I assume are just convention changes. The only thing is perhaps we should change the icon of the app.

**Comment by TobiTenno** [comment]
agreed, i'm working with yazu on one based off of the materials logo

**Comment by TobiTenno** [comment]
I'll rebase once I sort out the logo with yazu

**Comment by TobiTenno** [comment]
@EricSihaoLin any additional thoughts?

**Pull request merged**